### PR TITLE
skip NEG API call for NEG linker if possible

### PR DESCRIPTION
**Pull request by freehan**
This avoids unnecessary NEG API calls if possible. That would speed up ingress processing/

### CODE REVIEW

File: pkg/backends/neg_linker_test.go [hunk 1]
```diff
@@ -42,7 +43,7 @@ func newTestNEGLinker(fakeNEG negtypes.NetworkEndpointGroupCloud, fakeGCE *gce.C
 	(fakeGCE.Compute().(*cloud.MockGCE)).MockAlphaRegionBackendServices.UpdateHook = mock.UpdateAlphaRegionBackendServiceHook
 	(fakeGCE.Compute().(*cloud.MockGCE)).MockBetaRegionBackendServices.UpdateHook = mock.UpdateBetaRegionBackendServiceHook
 	(fakeGCE.Compute().(*cloud.MockGCE)).MockRegionBackendServices.UpdateHook = mock.UpdateRegionBackendServiceHook
-	return &negLinker{fakeBackendPool, fakeNEG, fakeGCE}
+	return &negLinker{fakeBackendPool, fakeNEG, fakeGCE, ctx.SvcNegInformer.GetIndexer()}
 }
 
 func TestLinkBackendServiceToNEG(t *testing.T) {
```
Comment (swetharepakula) [inline pkg/backends/neg_linker_test.go:49]: Can you add a test case that uses the svcneg informer?

**Comment by swetharepakula** [comment]
/approve

Overall lgtm. I think we should add one more test case for when the informer is populated.

File: pkg/backends/neg_linker_test.go (hunk 1 above)
Comment (freehan) [reply pkg/backends/neg_linker_test.go:51]: added test cases with svcneg Informer populated

File: pkg/backends/neg_linker_test.go [hunk 2]
```diff
@@ -233,7 +263,9 @@ func TestBackendsForNEG(t *testing.T) {
 					SelfLink:            "/neg1",
 				},
 			},
-			sp: &utils.ServicePort{},
+			sp: &utils.ServicePort{
+				VMIPNEGEnabled: true,
```
Comment (prameshj) [inline pkg/backends/neg_linker_test.go:267]: The L4 controller will set this to true when invoking negLinker.Link(), so this change is fine - https://github.com/kubernetes/ingress-gce/blob/80ac3c02662cde257307d1f48d6966c91127c6b8/pkg/l4/l4controller.go#L288

**Pull request merged**